src/py/flwr/client/app.py

```diff
@@ -450,6 +466,8 @@ def _on_backoff(retry_state: RetryState) -> None:
 
                     run.fab_id, run.fab_version = fab_id, fab_version
 
+                    supernode_tracker.record_run(runs[run_id])
```

Comment:
Would it make sense to move this line under the `if run_id not in runs` logic so we need to keep track of existing runs in the `SupernodeTracker` class? In that way the `record_run` will only be executed when there is a new run.

Reply:
I see your point. For now I would say let's leave this one, because it is a bit tricky to find the right place for doing the check `if run_id not in runs`. It can not be in line `468` because at that line, the `run_id` is already added to the `runs` and it won't be tracked.